darwin/dataset/download_manager.py

```diff
@@ -165,33 +176,85 @@ def download_image_from_annotation(
-    with annotation_path.open() as file:
-        annotation = json.load(file)
+    annotation = parse_darwin_json(annotation_path, count=0)
+    if annotation is None:
+        return None
 
     # If we are using folders, extract the path for the image and create the folder if needed
-    sub_path = annotation["image"].get("path", "/") if use_folders else "/"
+    sub_path = annotation.remote_path if use_folders else Path("/")
     parent_path = Path(image_path) / Path(sub_path).relative_to(Path(sub_path).anchor)
     parent_path.mkdir(exist_ok=True, parents=True)
 
-    if video_frames and "frame_urls" in annotation["image"]:
+    annotation.slots.sort(key=lambda slot: slot.name or "0")
+    if len(annotation.slots) > 0:
+        if force_slots:
+            _download_all_slots_from_json_annotation(annotation, api_key, parent_path, video_frames)
+        else:
+            _download_single_slot_from_json_annotation(annotation, api_key, parent_path, annotation_path, video_frames)
+
+
+def _download_all_slots_from_json_annotation(annotation, api_key, parent_path, video_frames):
+    for slot in annotation.slots:
+        slot_path = parent_path / sanitize_filename(annotation.filename) / sanitize_filename(slot.name)
+        slot_path.mkdir(exist_ok=True, parents=True)
+
+        if video_frames and slot.type != "image":
+            video_path: Path = slot_path / "sections"
+            video_path.mkdir(exist_ok=True, parents=True)
+            for i, frame_url in enumerate(slot.frame_urls or []):
+                path = video_path / f"{i:07d}.png"
+                _download_image(frame_url, path, api_key)
+        else:
+            for upload in slot.source_files:
+                file_path = slot_path / sanitize_filename(upload["file_name"])
+                _download_image(upload["url"], file_path, api_key)
+                _update_local_path(annotation, upload["url"], file_path)
+
+
+def _download_single_slot_from_json_annotation(annotation, api_key, parent_path, annotation_path, video_frames):
+    slot = annotation.slots[0]
+
+    if video_frames and slot.type != "image":
         video_path: Path = parent_path / annotation_path.stem
         video_path.mkdir(exist_ok=True, parents=True)
-        for i, frame_url in enumerate(annotation["image"]["frame_urls"]):
+        for i, frame_url in enumerate(slot.frame_urls or []):
             path = video_path / f"{i:07d}.png"
             _download_image(frame_url, path, api_key)
     else:
-        image_url = annotation["image"]["url"]
-        image_path = parent_path / sanitize_filename(annotation["image"]["filename"])
-        _download_image(image_url, image_path, api_key)
+        if len(slot.source_files) > 0:
+            image_url = slot.source_files[0]["url"]
+            image_path = parent_path / sanitize_filename(slot.filename or annotation.filename)
+            _download_image(image_url, image_path, api_key)
+            _update_local_path(annotation, image_url, image_path)
```

Comment:
why don't we loop over source files? the function only does the first slot, but if it consists of mutliple source files shouldn't they all be downloaded, `_update_local_path` seems to be able to handle it since it matches on `image_url` which will be unique per source_file.

Reply:
Are we fine with the fact that we'll get more files then items in target directory? We do loop over `source_files` when `--force_slots` option is used, because we also get a nice directory structure that groups source files per slot. Here we put everything in one spot, so it's gonna be a bit of a mess. 

But your call - if you confirm we want that either way - I'll do the change.